File: zero-rails/app/models/schema.rb
PersonType = GraphQL::ObjectType.define do
  name 'Person'
  description 'Somebody to lean on'

  field :id, !types.ID
  field :firstName, !types.String, property: :first_name
  field :lastName, !types.String, property: :last_name
  field :email, !types.String, 'Like a phone number, but spammier'
  field :username, !types.String, 'Use this to log in to your computer'
  field :friends, -> { types[PersonType] }, 'Some people to lean on'
  field :fullName do
    type !types.String
    description 'Every name, all at once'
    resolve -> (obj, args, ctx) { "#{obj.first_name} #{obj.last_name}" }
  end
end

QueryType = GraphQL::ObjectType.define do
  name 'Query'
  description 'The root of all queries'

  field :allPeople do
    type types[PersonType]
    description 'Everyone in the Universe'
    resolve -> (obj, args, ctx) { Person.all }
  end
  field :person do
    type PersonType
    description 'The person associated with a given ID'
    argument :id, !types.ID
    resolve -> (obj, args, ctx) { Person.find(args[:id]) }
  end
end

Schema = GraphQL::Schema.new(
  query: QueryType,
)

File: zero-django/zero_django/schema.py
import graphene

from people.models import Person

class PersonType(graphene.ObjectType):
    email = graphene.String(description='Like a phone number, but often longer')
    first_name = graphene.String()
    friends = graphene.List(lambda: PersonType, description='Mostly less strange people')
    full_name = graphene.String(description='Pretty much all of your name')
    id = graphene.String()
    last_name = graphene.String()
    username = graphene.String(description='Something you forget often')

    def resolve_friends(self, args, context, info):
        return self.friends.all()
    def resolve_full_name(self, args, context, info):
        return '{} {}'.format(self.first_name, self.last_name)

class QueryType(graphene.ObjectType):
    all_people = graphene.List(PersonType, description='A few billion people')
    person = graphene.Field(
        PersonType,
        id=graphene.ID(),
        description='Just one person belonging to an ID',
    )

    def resolve_all_people(self, args, context, info):
        return Person.objects.all()
    def resolve_person(self, args, context, info):
        id = args.get('id')
        return Person.objects.get(pk=id)

schema = graphene.Schema(query=QueryType)
